### PR TITLE
fix(dropdown): update option grouping logic

### PR DESCRIPTION
**Pull request by saty2103**
### What does this implement/fix? Explain your changes.

...

### Does this close any currently open issues?

...

### Any other comments?

...

### Dependent PRs/Commits

...

### Describe breaking changes, if any.

...

### Checklist

Check all those that are applicable and complete.

- [ ] Merged with latest `master` branch

### CODE REVIEW

File: core/components/atoms/dropdown/DropdownList.tsx [hunk 1]
```diff
@@ -539,10 +539,30 @@ const DropdownList = (props: OptionsProps) => {
...
+      acc[group].push(option); // Push the option into the group's array
+
+      return acc;
+    }, {} as { [key: string]: OptionSchema[] }); // Type asserting {} to provide type of key and value
+
+    const flattenedGroupList = Object.values(groupList).flatMap((item) => [...item]);
+
+    return flattenedGroupList;
+  };
+
```
Comment (anuradha9712) [inline core/components/atoms/dropdown/DropdownList.tsx]: Please move this inside `utils` file

**Review by anuradha9712 — CHANGES_REQUESTED** [review]
Please add a test case

File: core/components/atoms/dropdown/DropdownList.tsx [hunk 2]
```diff
@@ -539,10 +539,30 @@ const DropdownList = (props: OptionsProps) => {
...
+      const group = option.group || '';
+
+      if (!acc[group]) {
+        acc[group] = []; // Create an empty array for the group if it doesn't exist
+      }
+
+      acc[group].push(option); // Push the option into the group's array
+
+      return acc;
+    }, {} as { [key: string]: OptionSchema[] }); // Type asserting {} to provide type of key and value
```
Comment (anuradha9712) [inline core/components/atoms/dropdown/DropdownList.tsx]: Please remove the comments as it is very intuitive.

File: core/components/atoms/dropdown/DropdownList.tsx (hunk 1 above)
Comment (saty2103) [reply core/components/atoms/dropdown/DropdownList.tsx]: Done

File: core/components/atoms/dropdown/DropdownList.tsx (hunk 2 above)
Comment (saty2103) [reply core/components/atoms/dropdown/DropdownList.tsx]: Done

**Review by anuradha9712 — APPROVED** [review]

**Pull request merged**